CIP-XXX/CIP-XXX.md

```diff
@@ -27,10 +27,53 @@ As part of this, we also describe the bit ordering within a `BuiltinByteString`,
+1. Representing the empty set and the universe. We can use the `0x00`  or $00000000$ to represent an empty set, and `0xFF` or $11111111$ as the universe that can be represetend using $8$ bit byte string. 
+2. We can do set intersection using bit wise AND operation. Assume two sets $x = \{2,5,6\}$ and $y = \{2,4,5\}$, then $x \cap y = \{2,5\}$. The following example shows how we can represent set union using the bit wise AND operation.
+    ```mermaid 
+    graph LR 
+    A[00100110] --> C[Bitwise AND]
+    B[00101100] --> C
+    C --> D[00100100]
+    
+    ```
+3. We can do set union by using bit wise OR operation. Let us use the same sets $x$ and $y$. Then the unition $x \cup y = \{2,4,5,6\}$. The following image shows how the bitwise OR operation can be used to compute this union.
+    ```mermaid 
+    graph LR 
+    A[00100110] --> C[Bitwise OR]
+    B[00101100] --> C
+    C --> D[00101110]
+    
+    ```
+
+4. We can do complement using bit wise NOT (assuming finite universe). Let us take the set $x = \{2,5,6\}$, and assume that the universal set be $\{0,1,2,3,4,5,6,7\}$. Then, the complement of the set $x$, is $\neg x = \{0,1,3,4,7\}$. The following figure represents shows how we can represent these sets in byte strings and how bitwise NOT operation can be used to compute the complement of a set.
+    ```mermaid 
+    graph LR 
+    A[00100110] --> C[Bitwise NOT]
+    C --> D[11011001]
+    
+    ```
+
+5. We can also do set difference using XOR operation (when we are doing a symmetric difference). Just like previous examples we will take the sets $x$ and $y$. Then the symmetric difference between them would be $\{4,6\}$. As we can see in the next figure, the symmetric difference can be implemented using XOR bit wise operator.
+     ```mermaid 
+    graph LR 
+    A[00100110] --> C[Bitwise XOR]
+    B[00101100] --> C
+    C --> D[00001010]
+    ```
+6. Checking membership is as easy as checking an index of that `BuiltinByteString`. As per our definition, if we have $k$ in the set, then we are setting $k^{th}$ element in the byte string to represent the membership of $k$ in the integer set. So checking the membership for any element $k$ boils down to checking two things, (a) the size of the byte string and (b) the bit at the position $k$. 
+6. Adding and removing an element is as easy as setting and unsetting an index of that `BuiltinByteString`. For example if we want to add the element $1$ in the set $x$ then we only have to set the $1^{st}$ element of the bytestring representing the set $x$. On the other hand if we want to remove the emelent $2$ from the set $x$ then we will unset the $2^{nd}$ element from the set $x$.
+
+Clearly this is very efficient in terms of both space and time. 
```

Comment:
Why though? Efficiency in space and time needs to be justified relative what's already possible with existing work.